frontend/src/components/SamplesList.tsx

```diff
@@ -116,15 +124,42 @@ export default function SamplesList({
 
   const remoteCount = samples.length;
 
-  async function onCellValueChanged(
-    params: CellValueChangedEvent<SampleMetadataExtended>
-  ) {
+  async function onCellValueChanged(params: CellValueChangedEvent) {
     if (!editMode) return;
 
     const primaryId = params.data.primaryId;
     const fieldName = params.colDef.field!;
     const { oldValue, newValue, node: rowNode } = params;
 
+    // prevent registering a change if no actual changes are made
+    const noChangeInVal = rowNode.data[fieldName] === newValue;
+    const noChangeInEmptyCell =
+      _.isEmpty(rowNode.data[fieldName]) && _.isEmpty(newValue);
+    if (noChangeInVal || noChangeInEmptyCell) {
+      const updatedChanges = changes.filter(
+        (c) => !(c.primaryId === primaryId && c.fieldName === fieldName)
+      );
+      setChanges(updatedChanges);
+      if (updatedChanges.length === 0) setUnsavedChanges?.(false);
+      return;
+    }
```

Comment:
When a cell's database value is the same as the user's latest input, we don't want to register that as a "change" in the UI.

This is extra helpful for the Cost Center field to prevent the alert modal showing up when users click into a cell and exit it without making changes.